code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -60,108 +60,85 @@
+			to_chat(user,span("danger","\The [target.name] can't stomach anymore food!"))
+			return
+
+		amount_eaten = reagents.trans_to_mob(SA, min(reagents.total_volume,m_bitesize), CHEM_INGEST)
 
 		if (amount_eaten)
 			bitecount++
 			if (amount_eaten >= m_bitesize)
-				user.visible_message("<span class='notice'>[user] feeds [M] [src].</span>")
-				if (!istype(M.loc, /turf))//held mobs don't see visible messages
-					M << "<span class='notice'>[user] feeds you [src].</span>"
+				user.visible_message(span("notice","\The [user] feeds \the [target] \the [src]."))
+				if (!istype(target.loc, /turf))//held mobs don't see visible messages
+					to_chat(target,span("notice","\The [user] feeds you \the [src]."))
 			else
-				user.visible_message("<span class='notice'>[user] feeds [M] a tiny bit of [src]. <b>It looks full.</b></span>")
-				if (!istype(M.loc, /turf))
-					M << "<span class='notice'>[user] feeds you a tiny bit of [src]. <b>You feel pretty full!</b></span>"
-			On_Consume(M, user)
+				user.visible_message(span("notice","\The [user] feeds \the [target] a tiny bit of \the [src]. <b>It looks full.</b>"))
+				if (!istype(target.loc, /turf))
+					to_chat(target,span("notice","\The [user] feeds you a tiny bit of \the [src]. <b>You feel pretty full!</b>"))
 			return 1
+	else
+
+		var/is_full = 0
+		if(istype(target,/mob/living/carbon/human))
+			var/mob/living/carbon/human/H = target
+			var/limit = 550 * (1 + H.overeatduration / 2000)
+			var/current = H.nutrition + (H.reagents.get_reagent_amount("nutriment") * 25)
+			is_full = current >= limit
+
+		if(user == target)
+			if(!user.can_eat(src))
+				return
+			if(is_full)
+				to_chat(user,span("warning","You  can't stomach any more food!"))
+				return
+			reagents.trans_to_mob(target, min(reagents.total_volume,bitesize), CHEM_INGEST)
+			self_feed_message(user)
 		else
-			user << "<span class='danger'>[M.name] can't stomach anymore food!</span>"
+			if(!user.can_force_feed(target, src))
+				return
+			if(is_full)
+				to_chat(user,span("warning","\The [target] can't stomach any more food!"))
+				return
+
+			other_feed_message_start(user,target)
+			if(!do_mob(user, target))
+				return
+			other_feed_message_finish(user,target)
 
-	return 0
+			var/contained = reagentlist()
+			target.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has been fed [name] by [key_name(user)] Reagents: [contained]</font>")
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Fed [name] to [key_name(target)] Reagents: [contained]</font>")
+			msg_admin_attack("[key_name_admin(user)] fed [key_name_admin(target)] with [name] Reagents: [contained] (INTENT: [uppertext(user.a_intent)]) (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)",ckey=key_name(user),ckey_target=key_name(target))
+			reagents.trans_to_mob(target, min(reagents.total_volume,bitesize), CHEM_INGEST)
+
+	feed_sound(user)
+	bitecount++
+	On_Consume(target,user)

```

Comment:
Is this a new proc? If so, it should probably be `on_consume(target,user)` to match other procs. Snake case, all lowercase, or camel case.

Reply:
It's an old proc.